mne/coreg.py

```diff
@@ -168,6 +179,38 @@ def create_default_subject(mne_root=None, fs_home=None, update=False,
             shutil.copy(mne_fname % name, dest_bem)
 
 
+def create_high_res_head(subject, subjects_dir, fs_home):
+    """Wrapper for FreeSurfer $ mkheadsurf
+
+    Parameters
+    ----------
+    subject : str
+        Subject.
+    subjects_dir : str
+        Subjects directory.
+    fs_home : str
+        Path to the FreeSurfer installation.
+    """
+    if not os.path.exists(fs_home):
+        raise ValueError("Invalid fs_home parameter: %s does not exist" %
+                         fs_home)
+
+    env = os.environ.copy()
+    env['SUBJECTS_DIR'] = subjects_dir
+    env['SUBJECT'] = subject
+    env['FREESURFER_HOME'] = fs_home
+    env['PATH'] = os.path.join(fs_home, 'bin') + ':' + env['PATH']
+
+    subj_path = os.path.join(subjects_dir, subject)
+    if os.path.exists(os.path.join(subj_path, 'mri', 'T1.mgz')):
+        mri = 'T1.mgz'
+    else:
+        mri = 'T1'
+    run_subprocess('source $FREESURFER_HOME/FreeSurferEnv.sh; ' +
+                   'mkheadsurf -subjid ' + subject + ' -srcvol ' + mri,
+                   env=env, shell=True)
```

Comment:
I have no issue with the mne command, it does the same thing. :)

But what should the GUI do if the high resolution head is not present? Show an error message that asks the user to leave the GUI, run a command for each subject in the command line, and then restart the GUI? I feel like we discussed a similar issue with creating the average subject, where we ended up preferring that the GUI should automatically do a task for the user if it is deterministic... 



> On Jul 7, 2016, at 5:50 AM, Alexandre Gramfort <notifications@github.com> wrote:
> 
> In mne/coreg.py <https://github.com/mne-tools/mne-python/pull/3383#discussion_r69880037>:
> 
> > +                         fs_home)
> > +
> > +    env = os.environ.copy()
> > +    env['SUBJECTS_DIR'] = subjects_dir
> > +    env['SUBJECT'] = subject
> > +    env['FREESURFER_HOME'] = fs_home
> > +    env['PATH'] = os.path.join(fs_home, 'bin') + ':' + env['PATH']
> > +
> > +    subj_path = os.path.join(subjects_dir, subject)
> > +    if os.path.exists(os.path.join(subj_path, 'mri', 'T1.mgz')):
> > +        mri = 'T1.mgz'
> > +    else:
> > +        mri = 'T1'
> > +    run_subprocess('source $FREESURFER_HOME/FreeSurferEnv.sh; ' +
> > +                   'mkheadsurf -subjid ' + subject + ' -srcvol ' + mri,
> > +                   env=env, shell=True)
> I don't like this much as it means computation is done by the GUI while we should document how it should be done with mne command. If you don't agree with how the mne command does stuff the fix should be there.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/mne-tools/mne-python/pull/3383/files/5a207026f41c5c979f0480cc3294ceac3ef875fc#r69880037>, or mute the thread <https://github.com/notifications/unsubscribe/AAI5a9xUc4p948qh_gFaJxR5PSicOkNBks5qTMvKgaJpZM4JFP8_>.
>